### ISSUE TITLE
verify a single pin

### ISSUE THREAD

**Issue by sevenrats** [ipfs/kubo#9915]
### Checklist

- [X] My issue is specific & actionable.
- [X] I am not suggesting a protocol enhancement.
- [X] I have searched on the [issue tracker](https://github.com/ipfs/kubo/issues?q=is%3Aissue) for my issue.

### Description

Add the ability to verify the existence and integrity of a single pin by extending ```ipfs pin verify``` to accept a CID as an argument.  This would allow us to check the integrity of files we are concerned about without having to deal with the entire pinset, or waiting for the relatively long full verify procedure.

**Comment by arthurgavazza** [comment]
@Jorropo Can I pick this up?

**Comment by Jorropo** [comment]
@arthurgavazza yes :slightly_smiling_face: my guess is that you might have to do changes in [`boxo/pinning`](https://github.com/ipfs/boxo/tree/main/pinning) part of the codebase too.

Note, if you want something that would help us (kubo maintainers) https://github.com/ipfs/boxo/issues/327 would be nice, [`boxo/bitswap`](https://github.com/ipfs/boxo/tree/main/bitswap) is a critical piece of Kubo and most of the tests are skipped because they are really flaky and don't reliably run.

**Comment by aschmahmann** [comment]
Wouldn't the changes mostly live [here](https://github.com/ipfs/kubo/blob/50daf64be41903b4c7175ffb2d883d94ecf9cd88/core/commands/pin/pin.go#L600) and be relatively small (i.e. just run the `checkPin` function on one pin instead of all).

Also, while more convenient here AFAICT `ipfs dag stat --offline` accomplishes roughly the same task, although having a more specialized version in `pin` seems reasonable to me and easier for users to find.

**Comment by arthurgavazza** [comment]
Thx! I really need to improve my testing skills and it would be great to achieve that by contributing to such a core piece of Kubo.

**Comment by Jorropo** [comment]
@sevenrats sorry :heart: as @aschmahmann pointed out there are already ways to check if you have all the data from a pin, it's QOL.

**Comment by sevenrats** [comment]
if `ipfs dag stat --offline` verifies that the file is local and complete then thats just as easy for me to use.  we could even call this closed if you wanted.
